test/e2e/clusterctl_upgrade.go

```diff
@@ -69,11 +69,13 @@ type ClusterctlUpgradeSpecInput struct {
 	// provider contract to use to initialise the secondary management cluster, e.g. `v1alpha3`
 	InitWithProvidersContract string
 	SkipCleanup               bool
+	PreWaitForCluster         func(managementClusterProxy framework.ClusterProxy, managementClusterNamespace string, managementClusterName string)
```

Comment:
```suggestion
	PreWaitForCluster         func(managementClusterProxy framework.ClusterProxy, workloadClusterNamespace string, workloadClusterName string)
```

Just a nit, but I think from the POV of this function we are handing over the namespace+name of the workload cluster (and the proxy to the corresponding management cluster)

Reply:
Nice catch thanks